features/step_definitions/contracts/contract_steps.rb

```diff
@@ -1,21 +1,22 @@
-Then /^(?:the)? ((?:buyer|provider|account) "[^"]*") should have (?:a|an)(?: (pending|live|suspended))? (account|service|application) contract(?: with the (plan "[^"]*"))?$/ do |account, state, type , plan|
+# frozen_string_literal: true
+
+Then "(the )buyer/provider/{account} should have a/an( {word}) {word} contract with the {plan}" do |account, state, type, plan|
```

Comment:
Better maybe to use `pending/live/suspended` instead of `{word}`. Same with `account/service/application`.

I can't understand though how is `buyer/provider/{account}` supposed to work. Shouldn't it be `buyer/provider/account {string}`? Otherwise transformation would kick in on `account` and what's gonna happen in case it is `buyer` or `provider`?